reco_utils/dataset/movielens.py

```diff
@@ -42,73 +70,174 @@ def path(self):
     def has_header(self):
         return self._has_header
 
+    """ Item (Movie) file """
+
+    @property
+    def item_separator(self):
+        return self._item_sep
+
+    @property
+    def item_path(self):
+        return self._item_path
+
+    @property
+    def item_has_header(self):
+        return self._item_has_header
+
 
+# 10m and 20m data do not have user data
 _data_format = {
-    "100k": DataFormat("\t", "ml-100k/u.data"),
-    "1m": DataFormat("::", "ml-1m/ratings.dat"),
-    "10m": DataFormat("::", "ml-10M100K/ratings.dat"),
-    "20m": DataFormat(",", "ml-20m/ratings.csv", True),
+    "100k": _DataFormat(
+        "\t", "ml-100k/u.data", False,
+        "|", "ml-100k/u.item", False,
+    ),
+    "1m": _DataFormat(
+        "::", "ml-1m/ratings.dat", False,
+        "::", "ml-1m/movies.dat", False,
+    ),
+    "10m": _DataFormat(
+        "::", "ml-10M100K/ratings.dat", False,
+        "::", "ml-10M100K/movies.dat", False,
+    ),
+    "20m": _DataFormat(
+        ",", "ml-20m/ratings.csv", True,
+        ",", "ml-20m/movies.csv", True
+    ),
 }
 
-# Warning and error messages
-WARNING_MOVIE_LENS_HEADER = """MovieLens rating dataset has four columns (user id, movie id, rating, and timestamp)
-    but more than four column headers are provided. Will only use the first four column headers."""
-WARNING_HAVE_SCHEMA_AND_HEADER = (
-    "Both schema and header are provided. The header argument will be ignored."
+# 100K data genres index to string
+_genres = (
+    "unknown", "Action", "Adventure", "Animation",
+    "Children's", "Comedy", "Crime", "Documentary", "Drama", "Fantasy",
+    "Film-Noir", "Horror", "Musical", "Mystery", "Romance", "Sci-Fi",
+    "Thriller", "War", "Western"
 )
+
+# Warning and error messages
+WARNING_MOVIE_LENS_HEADER = """MovieLens rating dataset has four columns
+    (user id, movie id, rating, and timestamp), but more than four column headers are provided.
+    Will only use the first four column headers."""
+WARNING_HAVE_SCHEMA_AND_HEADER = """Both schema and header are provided.
+    The header argument will be ignored."""
 ERROR_MOVIE_LENS_SIZE = "Invalid data size. Should be one of {100k, 1m, 10m, or 20m}"
-ERROR_LOCAL_CACHE_PATH = (
-    "Local cache path only accepts a zip file path: use/something/like_this.zip"
-)
-ERROR_USER_ID_TYPE = "User id should be IntegerType"
-ERROR_MOVIE_ID_TYPE = "Movie id should be IntegerType"
-ERROR_RATING_TYPE = "Rating should be FloatType or DoubleType"
+ERROR_LOCAL_CACHE_PATH = """Local cache path only accepts a zip file path:
+    use/something/like_this.zip"""
 
 
-def load_pandas_df(size="100k", header=None, local_cache_path="ml.zip"):
+def load_pandas_df(
+    size="100k",
+    header=("UserId", "MovieId", "Rating", "Timestamp"),
```

Comment:
Question here: is there any relationship of these names `"UserId", "MovieId", "Rating", "Timestamp"` to those in this file https://github.com/Microsoft/Recommenders/blob/00bf677a714b7d1817c1e14115228b6724f0ad45/reco_utils/common/constants.py?